org.spoofax.jsglr2/src/main/java/org/spoofax/jsglr2/imploder/incremental/IncrementalTreeImploder.java

```diff
@@ -28,7 +28,7 @@
 
     private final IIncrementalImplodeInputFactory<ParseNode, Tree, Input> incrementalInputFactory;
     private final TreeImploder<ParseForest, ParseNode, Derivation, Tree, Input> regularImplode;
-    private final Map<String, WeakHashMap<ParseNode, SubTree<Tree>>> cache = new HashMap<>();
+    private final Map<FileObject, WeakHashMap<ParseNode, SubTree<Tree>>> cache = new HashMap<>();
```

Comment:
@mpsijm: @Gohla told me that `FileObject` does not guarantee `equals`/`hashCode` based on the file it represents. He said always to use `FileName` for keys in maps.

Reply:
Thanks for catching this, I did not know that! Will fix :slightly_smiling_face:

Reply:
Thanks @hendrikvanantwerpen.

Reference that confirms @mpsijm's [fix](https://github.com/metaborg/jsglr/pull/64/commits/501f87cbc93bc5731516cc4e91e8ea69e1e696bf) is safe: ["File names are immutable, and work correctly as keys in hash tables."](https://commons.apache.org/proper/commons-vfs/commons-vfs2/apidocs/org/apache/commons/vfs2/FileName.html).